LazyCache/Providers/MemoryCacheProvider.cs

```diff
@@ -33,6 +35,11 @@ public void Remove(string key)
             cache.Remove(key);
         }
 
+        public void RemoveAll()
+        {
+            cache = cacheFactory();
+        }

```

Comment:
Should the dispose method be called before renewing the MemoryCache or is that not needed?

Reply:
Yes, you're right, Dispose should be called to release the memory held by the previous cache instance.